protocols/grpc/hyperstart.proto

```diff
@@ -110,20 +116,55 @@ message CreateSandboxRequest {
 message DestroySandboxRequest {
 }
 
-message IpAddress {
-	string address = 1;
-	string mask = 2;
+enum IPFamily {
+	v4 = 0;
+	v6 = 1;
+}
+
+message IPAddress {
+	IPFamily family = 1;
+	string address = 2;
+	string mask = 3;
+}
+
+message Interface {
+	string device = 1;
+	string name = 2;
+	repeated IPAddress ipAddresses = 3;
+	uint64 mtu = 4;
+	string hwAddr = 5;
+}
+
+message Route {
+	string dest = 1;
+	string gateway = 2;
+	string device = 3;
+}
+
+enum UpdateType {
+	IP = 0;
+	Name = 1;
+	MTU = 2;
+}
+
+message AddInterfaceRequest {
+	Interface interface = 1;
+}
+
+message RemoveInterfaceRequest {
+	string name = 1;
 }
 
 message UpdateInterfaceRequest {
-	uint64 type	=1;
-	string device = 2;
-	string newName = 3;
-	repeated IpAddress ipAddresses = 4;
-	uint64 mtu = 5;
+	UpdateType type = 1;
+	Interface interface = 2;
+}
+
+message RouteRequest {
+	Route route = 1;
 }
 
-message AddRouteRequest {
+message UpdateRoutesRequest {
 	repeated Route routes = 1;
```

Comment:
Can we also update single route rule in one request? As for UpdateInterface

Reply:
@WeiZhang555 The idea with routes updating is that we always give it the full set of routes that we want to see applied (see @mcastelino [comment](https://github.com/kata-containers/agent/pull/4#issuecomment-347587537)), or else we'd have to follow the same logic as with the interface updating mecahnism (i.e. add an update type). I believe that routes are less independent than interfaces and should be updated in bulk.

Reply:
normally routing table are modified rule by rule, i.e. it should be ok if we provide two interface for adding and removing rule(s)

Reply:
Yes, I checked codes from hyperstart: https://github.com/hyperhq/hyperstart/blob/master/src/net.c#L294-L382 , route rules are handled one by one, so we don't need to pass all the route rules everytime, passing whole set of route rules is only behavior of runv in my thought.

and as @gnawux  said, "add" route and "remove" route should be enough for route operations, not sure if we need a "update" method since route is relatively simply comparing to network interface operations.

Reply:
Yes, I checked codes from hyperstart: https://github.com/hyperhq/hyperstart/blob/master/src/net.c#L294-L382 , route rules are handled one by one, so we don't need to pass all the route rules everytime, passing whole set of route rules is only behavior of runv in my thought.

and as @gnawux said, "add" route and "remove" route should be enough for route operations, not sure if we need a "update" method since route is relatively simply comparing to network interface operations.

Reply:
I'll let @mcastelino chime in. I'm fine with or without `UpdateRoutes`, and we can add it later on if we see a need for it.

Reply:
@WeiZhang555 Add and Delete assume that we are always stateful in both the runtime and the agent. The reason to have an update mechanism is to allow for a purely stateless setup of routes in this case, but for that matter anything else.  This will allow for stateless implementation and easier recovery.  An update in this case is an update of the full state, not necessarily of a single route (which does not make sense in any case). However we can add it later if  you feel that it is not necessary.